src/genegraph/transform/clinvar/cancervariants.clj

```diff
@@ -57,42 +68,68 @@
+            (log/debug :fn :vrs-allele-for-variation :body body)
             (-> body (get "variations") first wrap-with-canonical-variation add-vicc-context))
       ;; Error case
-      (log/error :fn ::normalize-general :msg "Error in VRS normalization request" :status status :response response))))
+      (log/error :fn :normalize-general :msg "Error in VRS normalization request" :status status :response response))))
 
 (defn normalize-canonical
   [^String variation-expression ^Keyword expression-type]
-  (log/info :fn ::normalize-canonical :variation-expression variation-expression)
-  (let [response (http-get-with-cache vicc-db-name
+  (log/info :fn :normalize-canonical :variation-expression variation-expression)
+  (let [response (http-get-with-cache vicc-db
                                       cancer-variants-url-to-canonical
                                       {:query-params {"q" variation-expression
                                                       "fmt" (name expression-type)}})
         status (:status response)]
     (case status
       200 (let [body (-> response :body json/parse-string)]
-            (log/debug :fn ::vrs-allele-for-variation :body body)
+            (log/debug :fn :normalize-canonical :body body)
             (-> body (get "canonical_variation") add-vicc-context))
       ;; Error case
-      (log/error :fn ::normalize-canonical :msg "Error in VRS normalization request" :status status :response response))))
+      (log/error :fn :normalize-canonical :msg "Error in VRS normalization request" :status status :response response))))
+
+(defn normalize-absolute-copy-number
+  [input-map]
+  (log/info :fn :normalize-absolute-copy-number :input-map input-map)
+  (let [response (http-get-with-cache vicc-db
+                                      absolute-copy-number-url
+                                      {:query-params
+                                       (into {}
+                                             (map #(vector (-> % first name) (-> % second))
+                                                  (select-keys input-map [:assembly
+                                                                          :chr
+                                                                          :start
+                                                                          :end
+                                                                          :total_copies])))})
+        status (:status response)]
+    (case status
+      200 (let [body (-> response :body json/parse-string)]
+            (log/info :fn :normalize-absolute-copy-number :body body)
+            (assert not-empty (-> body (get "absolute_copy_number")))
+            (-> body (get "absolute_copy_number") add-vicc-context))
+      ;; Error case
+      (log/error :fn :normalize-absolute-copy-number
+                 :msg "Error in VRS normalization request"
+                 :status status
+                 :response response))))
```

Comment:
Maybe?
```clojure
(defn normalize-absolute-copy-number
  "Normalize the absolute copy number in INPUT-MAP?"
  [input-map]
  (log/info :fn ::normalize-absolute-copy-number :input-map input-map)
  (let [url "https://normalize.cancervariants.org/variation/parsed_to_abs_cnv"
        {:keys [body status]}
        (-> input-map
            (select-keys [:assembly :chr :end :start :total_copies])
            (->> (map (fn [[k v]] [(name k) v]))
                 (into {})
                 (http/get url)))]
    (case status
      200 (let [{:strs [absolute_copy_number] :as body} (json/parse-string body)]
            (log/info :fn ::normalize-absolute-copy-number :body body)
            (assert not-empty absolute_copy_number)
            (add-vicc-context absolute_copy_number))
      (log/error :fn ::normalize-absolute-copy-number
                 :msg "Error in VRS normalization request"
                 :status status
                 :response response))))
```
… but I'd probably break it up into the service request vs the context addition.